src/WebJobs.Script/Config/TransmissionStatusHandler.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.Json.Serialization;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace Microsoft.Azure.WebJobs.Script.Config
+{
+    /// <summary>
+    /// Handler is used to track AI ingestion service response by subscribing to the transmission event.
+    /// The notification includes information about the response, and also the telemetry items which were part of the transmission.
+    /// </summary>
+    internal class TransmissionStatusHandler : IDisposable
+    {
+        private static readonly DiagnosticListener _source = new (ScriptConstants.HostDiagnosticSourcePrefix + "ApplicationInsights");
+
+        internal static void Handler(object sender, TransmissionStatusEventArgs args)
+        {
+            if (args == null)
+            {
+                _source.Write("TransmissionStatus", "Unable to parse transmission status response, args is null");
+                return;
+            }
+
+            // Always log if the response is non-success or if the feature flag is enabled
+            if (args.Response?.StatusCode != 200 || _source.IsEnabled(ScriptConstants.HostDiagnosticSourceDebugEventNamePrefix))
+            {
+                _source.Write("TransmissionStatus", FormattedLog(sender, args));
+            }
+        }
+
+        internal static string FormattedLog(object sender, TransmissionStatusEventArgs args)
+        {
+            var transmission = sender as Transmission;
+            var items = transmission?.TelemetryItems.GroupBy(n => n.GetEnvelopeName())
+                        .Select(n => new TelemetryItem
+                        {
+                            Type = n.Key,
+                            Count = n.Count()
+                        });
+
+            IngestionServiceResponse response = null;
+            if (!string.IsNullOrWhiteSpace(args?.Response?.Content))
+            {
+                response = JsonSerializer.Deserialize(
+                    args.Response.Content, IngestionServiceResponseContext.Default.IngestionServiceResponse);
+            }
+
+            string topErrorMessage = null;
+            int? topStatusCode = null;
+            if (response?.Errors != null && response.Errors.Length > 0)
+            {
+                topErrorMessage = response.Errors[0].Message;
+                topStatusCode = response?.Errors[0].StatusCode;
+            }
+
+            var log = new LogMessage
+            {
+                StatusCode = args?.Response?.StatusCode,
+                StatusDescription = args?.Response?.StatusDescription,
+                Id = transmission?.Id,
+                Items = items,
+                Received = response?.ItemsReceived,
+                Accepted = response?.ItemsAccepted,
```

Comment:
What do we want to do if response is null? Is this property okay to be null or empty?

Reply:
yes, response is null in case of success response. "Received" and "Accepted" won't be logged.